### PR TITLE
fix compact util type

### PR DESCRIPTION
**Pull request by michaltarasiuk**
Hello,
I fixed type in compact util.

I also changed the logic/createFormControl.ts and utils/get.ts, because the previous type of compact util influenced the returned result (functions return the same as before, I did not interfere with it).

I believe that the function itself should decide when it returns "any". Then the subsequent type change is easier if possible, because we know what causes the return type.

### CODE REVIEW

File: src/logic/createFormControl.ts [hunk 1]
```diff
@@ -484,7 +484,7 @@ export function createFormControl<
     return generateWatchOutput(names, _names, fieldValues, isGlobal);
   };
 
-  const _getFieldArray = (name: InternalFieldName) =>
+  const _getFieldArray = (name: InternalFieldName): any[] =>
```
Comment (bluebill1049) [inline src/logic/createFormControl.ts:487]: would it be possible to convert to `unknown` instead?

**Review by bluebill1049 — APPROVED** [review]
LGTM.

**Comment by bluebill1049** [comment]
thank you @MichalTarasiuk ❤️

File: src/logic/createFormControl.ts (hunk 1 above)
Comment (MichalTarasiuk) [reply src/logic/createFormControl.ts:487]: No, this function must return "any". The create Form Control function requires this. Function _getFieldArray before my changes also inference "any" type.

**Pull request merged**